src/components/InputTypes/Code.tsx

```diff
@@ -1,11 +1,37 @@
 /** @jsx jsx */
 import { jsx } from "@emotion/react";
-import React, { ChangeEvent } from "react";
+import React, { useEffect } from "react";
+
+import { basicSetup } from "@codemirror/basic-setup";
+import { python } from "@codemirror/lang-python";
+import { EditorState } from "@codemirror/state";
+import { oneDark } from "@codemirror/theme-one-dark";
+import { EditorView, ViewUpdate } from "@codemirror/view";
 
 interface CodeProps {
-    handler: (event: ChangeEvent<HTMLInputElement>) => void
+    handler: (newContent: string) => void,
+    questionId: string,
 }
 
 export default function Code(props: CodeProps): JSX.Element {
-    return <input type="text" className="text" onChange={props.handler}/>;
+    const onUpdate = () => EditorView.updateListener.of((v: ViewUpdate) => {
+        props.handler(v.state.doc.toString());
+    });
+
+    useEffect(() => {
+        const el = document.getElementById(`${props.questionId}-code`);
+        const state = EditorState.create({
+            extensions: [basicSetup, python(), onUpdate(), oneDark]
+        });
+        const view = new EditorView({
+            state,
+            parent: el as Element
+        });
+
+        return () => view.destroy();
+    }, []);
+
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
```

Comment:
These don't seem to be doing anything.

Reply:
I've run the TypeScript compiler both with and without and there's no difference, so I'll remove them.